vital/types/object_track_set.h

```diff
@@ -108,6 +110,17 @@ class VITAL_EXPORT object_track_state : public track_state
 
   detected_object_sptr detection;
 
+  point_2d_sptr& get_image_point() { return image_point_; }
+  const point_2d_cptr get_image_point() const { return std::const_pointer_cast<const point_2d>(image_point_); }
+
+  point_2d_sptr& get_track2d_point() { return track2D_point_; }
+  const point_2d_cptr get_track2d_point() const { return std::const_pointer_cast<const point_2d>(track2D_point_); }
+
+  point_3d_sptr& get_track3d_point() { return track3D_point_; }
+  const point_3d_cptr get_track3d_point() const { return std::const_pointer_cast<const point_3d>(track3D_point_); }
+
+  geo_point_sptr& get_geo_point() { return geo_point_; }
+  const geo_point_cptr get_geo_point() const { return std::const_pointer_cast<const geo_point>(geo_point_); }
```

Comment:
These lines seem extra long I would put them in the .cxx

Also odd that we have both an arbitrary 2d and 3d point in here; why not just a single 3d point?

I guess I'm against having both an image point and a bounding box; seems redundant

Reply:
I think this is a question for @mattbrown11 
I think he can explain the difference between the detection and the image point (if there is one...)

The tracking point can be 3D or 2D depending on the application coordinate system
I would not want to store a 2D cartesian point in a 3 value point...

Reply:
The reasoning behind including both the image_point and carrying around the original detection (with its bounding box therein) is
1) We may not always have or want to pass along the full detection (e.g., descriptor is big and expensive)
2) The bounding box encodes the spatial extent of the object being tracked, but we need to define a point on this object for tracking purposes. E.g., for tracking people from FMV, it makes sense to pull this point from the bottom center of the bounding box because that is where the person's feet are, and if you want to know where in the world a person in, you can unproject this image point from the camera model and intersect with your world surface model to figure out where in the world your person is standing. In WAMI, you typically want to pull image_point from the center of the bounding box. So, explicitly indicating which point within the bounding box you care about is important.
3) We would like some way to encode the location certainty for the object in the image coordinate system. If you are detecting people in FMV and you know your detector's bbox precision such that the covariance of your estimate for where the person's feet are is C, you'd like to encode this in the tracked image coordinate. This covariance get's used/transformed when mapping the image_point to tracking_point (e.g., meters) and geo_point.

Reply:
VIDTK also had a image_point in there too and I guess I don't care too much if it's here (I just won't use it, as long as it's still an optional sptr) - though I'd argue against the detection being big since it's usually just a box, it often doesn't contain a descriptor. I guess there could be a use case where you don't have a box but have just a image point, maybe.

Outside of the image_point, I'd still argue we don't need both a seperate 2d world (track?) point and a 3d world point and just need 1 - will likely get confusing otherwise and just like the geo point, the 3rd coordinate can be left blank for the 2d tracking case.

Reply:
Moving to one 3 dimensional track point